packages/one-bx-shell-app/src/app/components/iframe-launcher/iframe-launcher.component.ts

```diff
@@ -1,120 +1,119 @@
 import {
-  AfterViewInit,
   ChangeDetectionStrategy,
   Component,
   ElementRef,
   Input,
   OnChanges,
-  OnInit,
   Renderer2,
-  SimpleChanges,
-  ViewChild
+  ViewChild,
+  AfterViewInit
 } from '@angular/core';
-import {DomSanitizer, SafeUrl} from '@angular/platform-browser';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+import { LookupService } from '@fundamental-ngx/app-shell';
+
+interface MFFrameConfiguration {
+  name: string;
+  module?: string;
+  entry: string;
+  iframe: string;
+
+  scope?: string;
+  category?: string;
+  hasRoutes?: boolean;
+}
 
+interface MFAppConfiguration {
+  id: string;
+  provider?: string;
+  version: string;
+  host: string;
+  components: Record<string, MFFrameConfiguration>;
+}
 
-/**
- * This should be part of the plugin launcher and depending on the framework type we should either instantiate
- * remote all, or use directly this iFrame wrapper.
- *
- * IFrame launcher loads directly legacy application and they are not exposed as Module Federation plugin.
- * If these legacy application would have such capability there would be no need this component;
- *
- *
- *
- * We will extend PluginDescriptor with additional properties and if framework is Legacy we will handle this by iFrame
- * component. All this will be handled by
- *
- * <fdp-plugin-launcher name="fieldGlass.LandingLinks">
- * </fdp-plugin-launcher>
- *
- * This component can be used also as standalone
- *
- * Todo:
- *  - Finish fds-iframe-launcher standalone component, abstract whatever needs to be abstracted
- *  - Expand existing communication to some API
- *    AppShell -> iFrameLauncher -> iFrame contentWindow
- *    iFrame contentWindow -> iFrameLauncher -> the host app
- *   - Make it part of the plugin luncher and extend configuration.
- *
- *
- *
- *
- */
 @Component({
   selector: 'fds-iframe-launcher',
-  templateUrl: './iframe-launcher.component.html',
   styleUrls: ['./iframe-launcher.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <iframe
+      #iframe
+      class="responsive-wrapper"
+      [style.minHeight]="attrs.height"
+      load="onLoad(iframe)"
+      *ngIf="iconfig?.src"
+      [src]="iconfig.src"
+    ></iframe>
+  `
 })
-export class IFrameLauncherComponent implements OnInit, OnChanges, AfterViewInit {
-  @Input()
-  url: string;
-
-  @Input()
-  id: string;
-
-  @Input()
-  title: string;
+export class IframeLauncherComponent implements OnChanges, AfterViewInit {
+  @ViewChild('iframe', {static: false}) iframeEl: ElementRef;
+  iconfig: { src: string, attrs: Record<string, string> };
 
-  @Input()
-  height: string;
-
-  @Input()
-  attrs: Record<string, string | number>;
-
-  @ViewChild('iframe', {static: false})
-  iframeEl: ElementRef;
-
-  _iFrameUrl: SafeUrl;
+  @Input() url: string;
+  @Input() app: string;
+  @Input() component: string;
```

Comment:
What are the app and component @inputs goot for ?

Reply:
I don't want to delay this PR any longer, I will align naming across launchers accordingly to our discussion here https://github.com/valorkin/ng-mf-submodules/issues/26